client/src/components/pipelines/browser/Folder.js

```diff
@@ -894,6 +911,73 @@ export default class Folder extends localization.LocalizedReactComponent {
     await this.createPipeline({name, description, repository, token});
   };
 
+  createVersionedStorage = async ({
+    name,
+    description,
+    repository,
+    pipelineType
+  }) => {
+    const hide = message.loading(`Creating versioned storage ${name}...`, 0);
+    await this.createPipelineRequest.send({
+      name: name,
+      description: description,
+      parentFolderId: this._currentFolder.folder.id,
+      repository: repository,
+      pipelineType
+    });
+    hide();
+    if (this.createPipelineRequest.error) {
+      message.error(this.createPipelineRequest.error, 5);
+    } else {
+      this.closeCreateVersionedStorageDialog();
+      this.props.pipelines.fetch();
+      await this.props.folder.fetch();
+      if (this.props.onReloadTree) {
+        this.props.onReloadTree(true);
+      }
+    }
+  };
+
+  onCreateVersionedStorage = async ({name, description, repository}) => {
+    const pipelineType = 'VERSIONED_STORAGE';
+    if ((repository && repository.length)) {
+      const hide = message.loading('Checking repository existence...', -1);
+      await this.checkRequest.send({
+        repository
+      });
+      hide();
+      if (this.checkRequest.error) {
+        message.error(this.checkRequest.error);
+        return;
+      } else if (!this.checkRequest.value.repositoryExists) {
+        Modal.confirm({
+          title: 'Repository does not exist. Create?',
+          style: {
+            wordWrap: 'break-word'
+          },
+          content: null,
+          okText: 'OK',
+          cancelText: 'Cancel',
+          onOk: async () => {
+            await this.createVersionedStorage({
+              name,
+              description,
+              repository,
+              pipelineType
+            });
+          }
+        });
+        return;
+      }
+    }
+    await this.createVersionedStorage({
+      name,
+      description,
+      repository,
+      pipelineType
+    });
+  }
```

Comment:
@AleksandrGorodetskii this is mostly the same as `onCreatePipeline` method. We can refactor the code in the following way:
- create (extract logic from `onCreatePipeline`) a new method `checkRepositoryExistance(pipelineOpts, callback)` that checks repository existence and calls `callback(pipelineOpts)` if succeeded;
- use this method for creating a regular pipeline and for the versioned storage:
```javascript
/* pipeline creation modal callback */
<EditPipelineForm
          onSubmit={this.folderOperationWrapper(opts => this.checkRepositoryExistance(opts, this.createPipeline)}
/* versioned storage creation modal callback */
<VersionedStorageDialog
          onSubmit={this.folderOperationWrapper(opts => this.checkRepositoryExistance(opts, this.createVersionedStorage)}
```